Add cookiejar path related security issue

tirkarthi opened the pull request:
I had it in my fork and forgot to push it. bpo 30458 is also fixed but not updated. I will add them. I am opening this as WIP to make sure I don't forget it.

vstinner commented on vulnerabilities.yaml:
This one is already tracked.

vstinner commented on vulnerabilities.yaml:
```diff
@@ -1266,6 +1285,50 @@
+  disclosure: "2019-01-03 (Python issue bpo-35647 reported)"
+  reported-at: "2019-01-03"
+  reported-by: "Karthikeyan Singaravelan"
+  affected-versions:
```
Remove affected-versions, it's redundant.

vstinner commented on vulnerabilities.yaml:
```diff
@@ -21,6 +21,25 @@
 
+- name: "bpo-35755: Remove current directory from posixpath.defpath"
+  bpo: 35755
+  disclosure: "2019-01-17 (Reported by bpo-35755)"
```
disclosure, reported-at, reported-by and affected-versions are generated automatically.

vstinner commented on vulnerabilities.yaml:
```diff
@@ -21,6 +21,25 @@
 #
 # Other tag: "ignore-python3: true", used for modules removed from Python 3
 
+- name: "bpo-35755: Remove current directory from posixpath.defpath"
```
Honestly, I'm not sure that it's really a security issue. If you consider that it's the case, please open a discussion on security-sig@python.org mailing list.

tirkarthi replied:
I just added it since it was tagged as security in GitHub and bpo. I have removed it.

tirkarthi commented:
I have removed the auto-generated fields mentioned. I removed the urllib and defpath items as suggested. This PR adds only the cookiejar path related vulnerability.

vstinner commented on vulnerabilities.yaml:
```diff
@@ -1266,6 +1266,21 @@
     `http.cookiejar.DefaultCookiePolicy` policy.  Patch by Karthikeyan
     Singaravelan.
 
+- name: "bpo-35647: Incorrect validation of path in http.cookiejar"
```
Please remove "bpo-35647: " prefix.

vstinner commented on vulnerabilities.yaml:
```diff
@@ -1266,6 +1266,21 @@
+   - 3.8: 0e1f1f01058bd4a9b98cfe443214adecc019a38c
+  description: |
+    Cookies of example.com with path=/any where sent to example.com/anybad/
+    while using a cookiejar with http.cookiejar.DefaultCookiePolicy policy.
```
Please use reST syntax here for better rendering:

```
    Cookies of ``example.com`` with ``path=/any`` where sent to ``example.com/anybad/``
    while using a cookiejar with ``http.cookiejar.DefaultCookiePolicy`` policy.
```

vstinner commented:
Thanks, merged. It's good to see more contributors on this project ;-) I was feeling alone :-D

@tirkarthi: Can you try to add a script to detect if Python is vulnerable to bpo-35647 in check-python-vuln//check-python-vuln.py?

tirkarthi commented:
Thanks Victor for merging it. Sure, I will add a script as another PR.
